### PR TITLE
CDI shrinkwrap

### PR DESCRIPTION
**Pull request by licode**
Basic functions for Shrinkwrap algorithm in CDI is included in this PR. The aim of the current CDI code is for both 2D and 3D reconstruction.

As CSX will have data soon for image retrieval, CDI method needs to be ranked in high priority, per @stuwilkins 's request. I break the whole task into smaller PRs, and will send them soon.

### CODE REVIEW

**Comment by licode** [comment]
This work is a combination of my previous IDL code and the work of Xiaojing Huang in HXN. And this PR is ready to review.

**Comment by licode** [comment]
https://github.com/Nikea/scikit-xray/issues/258

**Comment by ericdill** [comment]
Looks good!  :+1:

**Comment by tacaswell** [comment]
What are the boundary condition differences? The `mode` switch does not cover the needed use-case?

**Comment by licode** [comment]
No, I tried different mode, and we usually use mode='same' for same size return.

**Comment by ericdill** [comment]
@tacaswell see https://github.com/Nikea/scikit-xray/issues/258

File: skxray/cdi.py [hunk 1]
```diff
@@ -94,3 +94,29 @@ def gauss(dims, sigma):
...
+def convolution(array1, array2):
+    """
+    Calculate convolution of two arrays. Transfer into q space to perform the calculation.
+
+    Parameters
+    ----------
+    array1 : array
+        The size of array1 needs to be normalized.
+    array2 : array
+        The size of array2 keeps the same
```
Comment (danielballan) [inline skxray/cdi.py]: keeps the same what?

**Pull request merged**